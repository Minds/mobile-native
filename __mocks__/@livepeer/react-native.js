// mock @livepeer/react-native
export const LivepeerConfig = {
  setConfig: jest.fn(),
  getConfig: jest.fn(),
  get: jest.fn(),
  set: jest.fn(),
};

export const createReactClient = jest.fn();
export const createLivepeerClient = jest.fn();
export const Player = jest.fn();
