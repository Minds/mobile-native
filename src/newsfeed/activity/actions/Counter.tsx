import React, { PureComponent } from 'react';

import { Text, View, TextStyle } from 'react-native';

import { CommonStyle as CS } from '../../../styles/Common';
import abbrev from '../../../common/helpers/abbrev';
import ThemedStyles from '../../../styles/ThemedStyles';

type PropsType = {
  size: number;
  count: number;
  testID?: string;
  style: TextStyle;
};

/**
 * Counters
 */
export default class Counter extends PureComponent<PropsType> {
  /**
   * Default Props
   */
  static defaultProps = {
    size: 15,
    style: undefined,
  };

  /**
   * Render
   */
  render(): React.ReactNode {
    const theme = ThemedStyles.style;
    const { count, style, ...otherProps } = this.props;

    return (
      <View style={CS.columnAlignCenter}>
        <Text
          style={[theme.colorIcon, theme.fontL, theme.fontMedium, style]}
          {...otherProps}>
          {count > 0 ? abbrev(count, 0) : ''}
        </Text>
      </View>
    );
  }
}
