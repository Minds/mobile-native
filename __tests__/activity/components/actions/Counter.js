import 'react-native';
import React from 'react';
import renderer from 'react-test-renderer';
import Counter from '../../../../src/newsfeed/activity/actions/Counter';

describe('Counter component', () => {
  let screen;
  beforeEach(() => {
    screen = renderer.create(<Counter count={100} />);

    jest.runAllTimers();
  });

  it('renders correctly', async () => {
    expect(screen).toMatchSnapshot();
  });

  it('should have Text', async () => {
    expect(screen.root.findAllByType('Text')).toHaveLength(1);
  });
});
