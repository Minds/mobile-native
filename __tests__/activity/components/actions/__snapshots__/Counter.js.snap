// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Owner component renders correctly 1`] = `
<View
  style={
    Object {
      "alignItems": "center",
      "flexDirection": "column",
    }
  }
>
  <Text
    size={15}
    style={
      Array [
        undefined,
        undefined,
        undefined,
        undefined,
      ]
    }
  >
    100
  </Text>
</View>
`;
